mathics/core/pattern.py

```diff
@@ -355,34 +361,34 @@ def yield_head(head_vars, _):
             # and not OneIdentity &
             # (expression.get_attributes(evaluation.definitions) |
             # expression.get_head().get_attributes(evaluation.definitions)):
-            new_expression = Expression(self.head, expression)
-            for leaf in self.leaves:
-                leaf.match_count = leaf.get_match_count()
-                leaf.candidates = [expression]
-                # leaf.get_match_candidates(
-                #    new_expression.leaves, new_expression, attributes,
+            new_expression = Expression(self.head.expr, expression)
```

Comment:
Here there is one of the three non trivial changes: `self.head` is a `Pattern` object, but `Expression` needs  a `BaseElement`.